healthcheck/healthcheck.go

```diff
@@ -80,15 +85,21 @@ func CreateVersionInfo(buildTime time.Time, gitCommit, version string) VersionIn
 	}
 }
 
-// AddClient adds a provided client to the healthcheck
-func (hc *HealthCheck) AddClient(c *Client) (err error) {
+// AddCheck adds a provided checker to the healthcheck
+func (hc *HealthCheck) AddCheck(checker *Checker) (err error) {
 	if hc.Started {
-		err = errors.New("unable to add new client, health check has already started")
-		log.Event(nil, "health check has already started", log.Error(err))
-		return
+		err := errors.New("unable to add new client, health check has already started")
+		return err
 	}
-	hc.Clients = append(hc.Clients, c)
-	return
+
+	client, err := newClient(checker)
```

Comment:
I think the function name `AddCheck` gives it away and we should be renaming the variable `client` to `check`

Reply:
I agree, the next PR will address this.  I was just worried about making too many changes all at once.